### ISSUE TITLE
Support promises

### ISSUE THREAD

**Issue by Jokero** [BryanDonovan/node-cache-manager#24]
Native promises are already supported in nodejs 0.12. I think all methods should return promise. And `wrap` method can take function returning promise:
```javascript
/**
 * @returns {Promise}
 */
function getCachedUser(id) {
    return memoryCache.wrap(id, function(/* optional cacheCallback */) {
        return getUser(id);
    }, ttl);
}
```

**Comment by BryanDonovan** [comment]
Thanks @Jokero - do you want to try implementing this?

**Comment by standy** [comment]
+1 on that feature

**Comment by BryanDonovan** [comment]
I don't use promises myself, so if someone else wants to try implementing support for them, that would be great.

**Comment by cgmartin** [comment]
In case it helps anyone, here's an example workaround using bluebird's `promisifyAll()` with cache-manager's `wrap()` function: https://gist.github.com/cgmartin/2df6f68d08388bc8c398

(The first `work/cacheMiss` callback doesn't precisely match @Jokero's original example, but it gets close. I'd be curious if that could be improved upon.)

I'm not sure if it was explicitly called out, but if promise support was added consider making it optional for those who don't use promises... using something like [bluebird's `.asCallback()`](http://bluebirdjs.com/docs/api/ascallback.html)
> This can be used to create APIs that both accept node-style callbacks and return promises

**Comment by jkernech** [comment]
+1

**Comment by BryanDonovan** [comment]
It does seem like using `Promise.promisifyAll` or equivalent would work, right? I could export a "promisified" version of the module perhaps.  I'm not sure what library to use though (I don't see built-in promises in node 0.12.x).

**Comment by marttiku** [comment]
My solution for the problem using bluebird.

Setup:

```javascript
var cacheManager = require('cache-manager');
var cache = cacheManager.caching({
    store: 'memory',
    max: 100,
    ttl: 30 /*seconds*/
});

bluebird.promisifyAll(cache);

cache.promise = function (cacheKey, work, ttl) {
	ttl = ttl || 10;
    return cache.wrapAsync(cacheKey, function(cacheResult) {
            return work().asCallback(cacheResult);                   
        }, {
            ttl: ttl
        }
    );	
}
```

and usage:

```javascript
            var cachekey = 'session_' + req.token;
            var ttl = 10;

            function work() {
                return models.Session.findOne({
                    where: {
                        token: req.token
                    }
                })                
            }

     var promise =  cache.promise(cachekey, work, ttl);
```

**Comment by PuKoren** [comment]
Currently doing a PR, #39

I have issues with the get method in single store cache usage: either the cache store has to implement the promise support, or cache-manager has to add a breaking change to handle this in a generic way for every available store.

Any suggestion is welcome

**Comment by BryanDonovan** [comment]
Closed by #39

**Comment by jkernech** [comment]
Thanks @PuKoren !!!
